modules/references/client/components/ListReferences.component.js

```diff
@@ -14,28 +14,61 @@ import ReferencesSection from './read-references/ReferencesSection';
  */
 export default function ListReferences({ profile, authenticatedUser }) {
   const { t } = useTranslation('references');
-  const [publicReferences, setPublicReferences] = useState([]);
-  const [pendingReferences, setPendingReferences] = useState([]);
+  const [publicExperiencePairs, setPublicExperiencePairs] = useState([]);
+  const [pendingExperiences, setPendingExperiences] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
   async function fetchReferences() {
     setIsLoading(true);
     try {
-      const references = await readReferences({
+      const experiences = await readReferences({
         userTo: profile._id,
+        includeReplies: true,
       });
 
-      const publicNewestFirst = references.filter(
-        reference => reference.public,
+      const publicExperiences = experiences.filter(
+        experience => experience.public,
       );
-
-      const pendingNewestFirst = references.filter(
-        reference => !reference.public,
+      const pendingNewestFirst = experiences.filter(
+        experience => !experience.public,
       );
+
+      /* TODO: compare by ids (couldn't make it work, so I used usernames to compare for now) */
```

Comment:
Just leaving a note here to have a look. :-)

Reply:
When we fetch references, looks like ids are not fetched with `userTo` and `userFrom` objects:

![image](https://user-images.githubusercontent.com/2955794/101295817-3557fd80-3820-11eb-8f93-8b97774dc564.png)

Is there anything wrong with $aggregate query and $project thing? (haven't looked much into mongoose yet)